core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -260,60 +262,48 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         } else {
             builder.field(TYPE, getExceptionName());
             builder.field(REASON, getMessage());
-            for (String key : headers.keySet()) {
-                if (key.startsWith("es.")) {
-                    List<String> values = headers.get(key);
-                    xContentHeader(builder, key.substring("es.".length()), values);
+
+            Set<String> customHeaders = null;
+            for (String header : headers.keySet()) {
```

Comment:
would you mind reverting the variable name change, at least temporarily? it confuses the review